### ISSUE TITLE
group_by function 

### ISSUE THREAD

**Issue by akhst7** [krassowski/complex-upset#66]
I have data with a group column as follows;

```
str(category)
     A B C D E F  G  H cluster
  1: 0 0 0 0 1 1  0  1      10
  2: 0 0 0 0 0 0  0  1       6
  3: 0 0 0 1 0 0  1  0       8
  4: 0 0 0 0 0 0  0  0       9
  5: 0 1 0 0 1 1  0  1       9
 ---                          
170: 0 1 0 0 0 0  0  0       4
171: 0 1 0 1 0 0  0  0       6
172: 0 0 0 0 1 0 NA NA       2
173: 0 0 0 1 0 0  1  0       7
174: 0 1 0 1 0 0  0  0       9
```
I would like to show the upset plot grouped by cluster.  I played with `sort_intersection `but did not quite work.  Is there a function similar to UpsetR's implementation of` group_by`  ?

Thanks.

**Comment by krassowski** [comment]
Good idea! I added `group_by` argument for feature parity. It is available in the v0.5.19 - you can update the package using:

```
devtools::install_github("krassowski/complex-upset")
```

Please see the example here: https://krassowski.github.io/complex-upset/articles/Examples_R.html#9-grouping-intersections

Something like this is now easily possible:
![image](https://user-images.githubusercontent.com/5832902/98469957-771e6580-21da-11eb-8b06-49646c06b69f.png)

**Comment by akhst7** [comment]
@krassowski 

Is it possible to manually define "sets" ?  For above example, use "cluster" as a grouping value, like this` group_by(cluster)`.

**Comment by krassowski** [comment]
Not currently, but i will look into this.

**Comment by krassowski** [comment]
FYI, the performance of `group_by` got a notable boost with e6814c160a5610a7aa02787141e52965f6f209bf and 866132729d6fbbe0a89f35a380d1af3f1a951b09 (version 0.7.4), roughly from 16.7 s to 1.7 s for ~1000 observations x 70 sets. Also, the 16.7 s is *after* a bunch of optimizations not specific to this option, so the actual improvement is even better.
